packages/synchro-charts/src/components/sc-dial/sc-dial-base/sc-dial-base.tsx

```diff
@@ -0,0 +1,186 @@
+import { Component, h, Prop, State, Watch } from '@stencil/core';
+import { DataPoint, DataStream, MinimalSizeConfig, Primitive, ViewPortConfig } from '../../../utils/dataTypes';
+import { round } from '../../../utils/number';
+import { isNumberDataStream } from '../../../utils/predicates';
+import { getIcons } from '../../charts/common/annotations/iconUtils';
+import { Threshold } from '../../charts/common/types';
+import { NO_VALUE_PRESENT } from '../../common/terms';
+import { DialLoading } from './sc-dial-loading';
+import { TextSizeConfig, sizeContent, StatusProgress, LengthToSize, SizeFont } from './util';
+
+const title = (dataStream: { detailedName?: any; name?: any } | null | false) => {
+  if (dataStream) {
+    return dataStream.detailedName || dataStream.name;
+  }
+  return null;
+};
+
+const UNITMINLENGTH = 1;
+@Component({
+  tag: 'sc-dial-base',
+  styleUrl: 'sc-dial-base.css',
+  shadow: false,
+})
+export class ScDialBase {
+  @Prop() viewport: ViewPortConfig;
+  @Prop() breachedThreshold?: Threshold;
+  @Prop() valueColor?: string; // css color string
+
+  @Prop() alarmStream?: DataStream;
+
+  @Prop() propertyStream?: DataStream;
+  @Prop() propertyPoint?: DataPoint<Primitive>;
+  @Prop() size?: MinimalSizeConfig = { width: 0, height: 0 };
+
+  @Prop() isLoading?: boolean = false;
+
+  @State() fontSize: TextSizeConfig = sizeContent.XL;
+
+  getColorByPercent = (percent: number): string => {
```

Comment:
This is not how thresholds should be determined. There are thresholds contained within the `annotations` field which would be set to declare thresholds, which are used to get a color

Reply:
The getColorByPercent method has been removed from passing in color values externally.